debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java

```diff
@@ -178,7 +178,7 @@ protected abstract SnapshotResult<O> doExecute(ChangeEventSourceContext context,
     /**
      * Prepares the taking of a snapshot and returns an initial {@link SnapshotContext}.
      */
-    protected abstract SnapshotContext<P, O> prepare(P partition) throws Exception;
+    protected abstract SnapshotContext<P, O> prepare(P partition, boolean blocking) throws Exception;
```

Comment:
@mfvitale I think the parameter here and elswhere should be renamed to either `adHoc` or `onDemand` to better reflect the situation. The initial snapshot is always blocking in the sense that streaming it not running.

Reply:
Good point.